### PR TITLE
Duo Security Connector

### PR DESCRIPTION
**Pull request by dicolanl**
Fixes #

## Proposed Changes
Includes:
Function Connector
Parsers
Detections
Workbook

### CODE REVIEW

File: Detections/DuoSecurity/IPEntity_DuoSecurity.yaml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+      | where TimeGenerated >= ago(dt_lookBack)
+      | where isnotempty(access_device_ip_s)
+      // renaming time column so it is clear the log this came from
+      | extend Duo_TimeGenerated = isotimestamp_t
+  )
+  on $left.TI_ipEntity == $right.access_device_ip_s
+  | summarize LatestIndicatorTime = arg_max(TimeGenerated, *) by IndicatorId
+  | project LatestIndicatorTime, Description, ActivityGroupNames, IndicatorId, ThreatType, Url, ExpirationDateTime, ConfidenceScore, Duo_TimeGenerated,
+  TI_ipEntity, user_name_s, facto_s, result_s, application_name_s, event_type_s, txid_g, user_key_s, access_device_ip_s, access_device_location_city_s, access_device_location_state_s, access_device_location_country_s
+  | extend timestamp = Duo_TimeGenerated, IPCustomEntity = access_device_ip_s, AccountCustomEntity = user_name_s,
```
Comment (shainw) [inline Detections/DuoSecurity/IPEntity_DuoSecurity.yaml:47]: Drop the comma at the end of this line

File: Detections/DuoSecurity/TrustMonitorEvent.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+queryPeriod: 5m
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - CredentialAccess
+query: |
+
+  let timeframe = ago(5m);
+  DuoSecurityTrustMonitor_CL
+  | where TimeGenerated >= timeframe
```
Comment (shainw) [inline Detections/DuoSecurity/TrustMonitorEvent.yaml:20]: Should there be a filter here for a certain type of event or is everything in this table something that should be alerted on?

**Review by shainw — CHANGES_REQUESTED** [review]
I don't see sample data set included which is required for parser testing.

File: Detections/DuoSecurity/IPEntity_DuoSecurity.yaml (hunk 1 above)
Comment (dicolanl) [reply Detections/DuoSecurity/IPEntity_DuoSecurity.yaml:47]: New commit is fixed

File: Detections/DuoSecurity/TrustMonitorEvent.yaml (hunk 2 above)
Comment (dicolanl) [reply Detections/DuoSecurity/TrustMonitorEvent.yaml:20]: These are detections in Duo Security (https://duo.com/docs/trust-monitor) they do monitoring of their logs and generate alerts in Trust Monitor so they should be investigated in Sentinel

**Comment by dicolanl** [comment]
> I don't see sample data set included which is required for parser testing.

I have added via a new commit

**Comment by preetikr** [comment]
@nazang / @v-jayakal - please review the data connector and workbook
@shainw for parser + detections.

**Review by v-maudan — CHANGES_REQUESTED** [review]
@dicolanl Please update the following requested changes:

1. Please move these files to [Solution folder](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions). Follow this [folder structure](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/Box) for reference.
2. Please update changes related to Azure gov. Refer to this [PRLink ](https://github.com/Azure/Azure-Sentinel/pull/1762/files)for gov changes.

File: Parsers/DuoSecurity/DuoSecurityAdministrator.txt [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+// Last Updated:    02/11/2021
+// Comment:         Inital Release
+//  
+// DESCRIPTION:
+// This parser takes Duo Security Administrator records ingested from the Function App Data Connector and parses the data into a normalized schema
+//
+// USAGE:
+// 1. Open Log Analytics/Azure Sentinel Logs blade. Copy the query below and paste into the Logs query window. 
+// 2. Click the Save button above the query. A pane will appear on the right, select "as Function" from the drop down. Enter a Function Name.
+//    It is recommended to name the Function Alias, as DuoSecurity<Log>
```
Comment (v-maudan) [inline Parsers/DuoSecurity/DuoSecurityAdministrator.txt:13]: @dicolanl , Can you please let us know where we used these parsers.

File: Parsers/DuoSecurity/DuoSecurityAuthentication.txt [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+// Last Updated:    02/11/2021
+// Comment:         Inital Release
+//  
+// DESCRIPTION:
+// This parser takes Duo Security Authentication records ingested from the Function App Data Connector and parses the data into a normalized schema
+//
+// USAGE:
+// 1. Open Log Analytics/Azure Sentinel Logs blade. Copy the query below and paste into the Logs query window. 
+// 2. Click the Save button above the query. A pane will appear on the right, select "as Function" from the drop down. Enter a Function Name.
+//    It is recommended to name the Function Alias, as DuoSecurity<Log>
```
Comment (v-maudan) [inline Parsers/DuoSecurity/DuoSecurityAuthentication.txt:13]: Give proper function alias to save parser.[All parser]

File: DataConnectors/DuoSecurity/DuoSecurity_API_FunctionApp.json [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+        }
+    ]
+}
```
Comment (v-maudan) [inline DataConnectors/DuoSecurity/DuoSecurity_API_FunctionApp.json:185]: Please add a Metadata object within this file. See [https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Connector_REST_API_AzureFunctionApp_template/DataConnector_API_AzureFunctionApp_template.json](https://github.com/Azure/Azure-Sentinel/blob/master/DataConnectors/Templates/Connector_REST_API_AzureFunctionApp_template/DataConnector_API_AzureFunctionApp_template.json) for further detail.

**Review by v-maudan — CHANGES_REQUESTED** [review]

File: DataConnectors/DuoSecurity/DuoSecurity_API_FunctionApp.json [hunk 6]
```diff
@@ -0,0 +1,185 @@
+{
+    "id": "DuoSecurity",
+    "title": "Duo Security (Preview)",
```
[the remark this answers is not in the document]
Comment (dicolanl) [reply DataConnectors/DuoSecurity/DuoSecurity_API_FunctionApp.json]: Done

File: Parsers/DuoSecurity/DuoSecurityAuthentication.txt (hunk 4 above)
Comment (dicolanl) [reply Parsers/DuoSecurity/DuoSecurityAuthentication.txt:13]: @v-maudan  do you have an example?

**Comment by dicolanl** [comment]
> @dicolanl Please update the following requested changes:
> 
> 1. Please move these files to [Solution folder](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions). Follow this [folder structure](https://github.com/Azure/Azure-Sentinel/tree/master/Solutions/Box) for reference.
> 2. Please update changes related to Azure gov. Refer to this [PRLink ](https://github.com/Azure/Azure-Sentinel/pull/1762/files)for gov changes.

I have moved to solutions folder and added support for Gov

**Comment by v-maudan** [comment]
@dicolanl , Can you please fix PR validation errors. Thanks!

File: Parsers/DuoSecurity/DuoSecurityAuthentication.txt (hunk 4 above)
Comment (dicolanl) [reply Parsers/DuoSecurity/DuoSecurityAuthentication.txt]: @v-jayakal the parsers are not currently used in the workbooks or analytics.  They are available if the customer wants to use them

**Comment by dicolanl** [comment]
> @dicolanl , Can you please fix PR validation errors. Thanks!

@v-maudan I created a new PR #2422 with CL schemas.  Need this approved and merged to fix validations!

**Comment by dicolanl** [comment]
@v-maudan All checks now pass validation

**Pull request merged**